index_configuration_tool/main.py

```diff
@@ -12,6 +12,18 @@
 HOSTS_KEY = "hosts"
 USER_KEY = "username"
 PWD_KEY = "password"
+INSECURE_KEY = "insecure"
+CONNECTION_KEY = "connection"
+
+
+# This config key may be either directly in the main dict (for sink)
+# or inside a nested dict (for source). The default value is False.
+def is_insecure(config: dict) -> bool:
+    if INSECURE_KEY in config:
+        return bool(config[INSECURE_KEY])
```

Comment:
at least based on the examples you have in the tests, why do you need to type-convert this to `bool`? It looks like it's a bool already, but if there's a chance it isn't, type-converting it can have unexpected effects (e.g. `bool("False") == True`.

Reply:
Good catch, fixed